core/internal/sentry_ext/client.go

```diff
@@ -0,0 +1,171 @@
+	if err != nil {
+		slog.Error("sentry_ext: New: failed to create cache", "err", err)
+		return nil
+	}
+
+	// If the DSN is not set, the client is effectively disabled.
+	return &Client{
+		Recent: cache,
+	}
+}
+
+// SetUser sets the user information for the sentry client.
+func (s *Client) SetUser(id, email, name string) {
+	sentry.ConfigureScope(func(scope *sentry.Scope) {
+		scope.SetUser(sentry.User{
+			ID:    id,
+			Email: email,
+			Name:  name,
+		})
+	})
+}
+
+// CaptureException captures an error and sends it to sentry.
+// Used for capturing errors. The error is sent to sentry as an error level
+// event. The event is enriched with the tags provided.
+func (s *Client) CaptureException(err error, tags map[string]string) {
+	if !s.Recent.shouldCapture(err) {
+		return
+	}
+
+	// Send the error to sentry
+	localHub := sentry.CurrentHub().Clone()
+	localHub.ConfigureScope(
+		func(scope *sentry.Scope) {
+			scope.SetTags(tags)
+		},
+	)
+	localHub.CaptureException(err)
+}
+
+// CaptureMessage captures a message and sends it to sentry.
+// Used for capturing non-error messages. The message is sent to sentry as an
+// info level event. The event is enriched with the tags provided.
+func (s *Client) CaptureMessage(msg string, tags map[string]string) {
+	if !s.Recent.shouldCapture(errors.New(msg)) {
+		return
+	}
+
+	localHub := sentry.CurrentHub().Clone()
+	localHub.ConfigureScope(
+		func(scope *sentry.Scope) {
+			scope.SetTags(tags)
+		},
+	)
+	localHub.CaptureMessage(msg)
+}
+
+// Reraise captures an error and re-raises it.
+// Used to capture unexpected panics.
+func (s *Client) Reraise(err any, tags map[string]string) {
+	if err != nil {
+		var e error
+		if errors.As(e, &err) {
+			s.CaptureException(e, tags)
+		} else {
+			e = fmt.Errorf("%v", err)
+			s.CaptureException(e, tags)
+		}
+		sentry.Flush(time.Second * 2)
+		panic(err)
+	}
+}
+
+// Flush flushes the sentry client.
+func (s *Client) Flush(timeout time.Duration) bool {
+	hub := sentry.CurrentHub()
+	return hub.Flush(timeout)
+}
+
+// RemoveBottomFrames modifies the stack trace by checking the file name of the bottom-most 3 frames
+// and removing them if they are internal to core
+func RemoveBottomFrames(event *sentry.Event, hint *sentry.EventHint) *sentry.Event {
+	for i, exception := range event.Exception {
+		if exception.Stacktrace == nil {
+			continue
+		}
+		frames := exception.Stacktrace.Frames
+		framesLen := len(frames)
+		// for the recovered panics, the bottom-most 3 frames of the stacktrace
+		// will come from sentry.go and logging.go, so we remove them
+		if framesLen < 3 {
+			continue
+		}
+		for j := framesLen - 1; j >= framesLen-3; j-- {
+			frame := frames[j]
+			// TODO: think of a better way to do this without hard-coding the file names
+			//  this is a hack to remove the bottom-most 3 frames that are internal to core
+			if strings.HasSuffix(frame.AbsPath, "sentry.go") || strings.HasSuffix(frame.AbsPath, "logging.go") {
```

Comment:
make sure to adjust this to the new file names

Reply:
will fix in a follow up.

Reply:
https://github.com/wandb/wandb/pull/7886